velox/functions/prestosql/Bitwise.h

```diff
@@ -95,67 +95,79 @@ struct BitwiseArithmeticShiftRightFunction {
   }
 };
 
-namespace {
-template <typename T, int MAX_SHIFT>
+namespace detail {
+template <typename T>
 #if defined(__clang__)
 __attribute__((no_sanitize("integer")))
 #endif
-FOLLY_ALWAYS_INLINE bool
-bitwiseLeftShift(int64_t& result, T number, T shift) {
-  if ((uint64_t)shift >= MAX_SHIFT) {
-    result = 0;
+FOLLY_ALWAYS_INLINE T
+bitwiseLeftShift(T number, int32_t shift) {
+  static constexpr uint32_t kMaxShift = sizeof(T) * 8;
+  if ((uint32_t)shift >= kMaxShift) {
```

Comment:
what happens here if shift was negative? Is the expectation that this function will just return zero? 

Maybe just add a quick comment above?

Reply:
Adding "// Return zero if 'shift' is negative or exceeds number of bits in T."